api/repositories/app_repository_test.go

```diff
@@ -330,163 +334,180 @@ var _ = Describe("AppRepository", func() {
 
 	Describe("CreateApp", func() {
 		const (
-			defaultNamespace = "default"
+			testAppName = "test-app-name"
+		)
+		var (
+			appCreateMessage AppCreateMessage
+			spaceGUID        string
 		)
 
-		When("creating an App record and", func() {
-			const (
-				testAppName = "test-app-name"
-			)
+		BeforeEach(func() {
+			spaceGUID = generateGUID()
 
-			var (
-				testAppGUID    string
-				emptyAppRecord = AppRecord{}
-			)
+			Expect(k8sClient.Create(testCtx, &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{Name: spaceGUID},
+			})).To(Succeed())
+
+			appCreateMessage = initializeAppCreateMessage(testAppName, spaceGUID)
+		})
+
+		AfterEach(func() {
```

Comment:
It looks like there is a test failing due to some test pollution related to this repo. I'm wondering if we're missing a delete `"test-app-name"` call here in this `AfterEach`?

Reply:
My theory has been that we should be able to just delete the namespace that contains the resources, and it will cascade the delete to the resources in the namespace. However, this doesn't seem to work consistently in test, which is confusing.